mobilecoind/api/src/lib.rs

```diff
@@ -21,6 +21,15 @@ mod autogenerated_code {
 
 pub use autogenerated_code::{mobilecoind_api::*, *};
 
+// This is used for some tests, we want to test that the fields that appeared
+// in an outlay in a request match the corresponding fields in the outlay in the
+// response.
+impl PartialEq<Outlay> for OutlayV2 {
+    fn eq(&self, other: &Outlay) -> bool {
+        self.value == other.value && self.receiver == other.receiver
```

Comment:
I think this also must check for self.token_id == 0

Reply:
So I think in some cases, there are tests that are like, call `generate_tx`, where `token_id = 1`, and `OutlayV1` is used.

Then, in the `TxProposal` that results, the outlays that are recorded are now an `OutlayV2`, and they do have all their token id's stored with them, so they will have `token_id = 1`.

So if I also check that `self.token_id == 0`, then that test will be failing.

---

Probably the right thing to do is, not create this PartialEq impl, and instead go fix up those tests so that when comparing Outlay and OutlayV2, we convert Outlay to OutlayV2 first providing appropriate context. I did it this way first because it was faster to develop.

Reply:
Thats also acceptable (not creating it)